testsuite/bond_tests.py

```diff
@@ -134,11 +141,13 @@ def test_bond_fene(self):
         bond = {'k'      : 400 * pmb.units('reduced_energy / reduced_length**2'),
                 'd_r_max': 0.8 * pmb.units.nm}
 
-        with contextlib.redirect_stdout(io.StringIO()) as f:
-            pmb.define_bond(bond_type = bond_type,
+        pmb.define_bond(bond_type = bond_type,
                             bond_parameters = bond,
                             particle_pairs = [['A', 'A']])
-            self.assertEqual(f.getvalue(), 'WARNING: No value provided for r_0. Defaulting to r_0 = 0\n')
+        log_contents = log_stream.getvalue()
+        assert "no value provided for r_0. Defaulting to r_0 = 0" in log_contents
```

Comment:
here and in subsequent test methods, consider using `self.assertIn()` instead of `assert`